packages/server/src/ApolloServer.ts

```diff
@@ -1058,28 +1048,24 @@ export class ApolloServer<in out TContext extends BaseContext = BaseContext> {
    * just a convenience, not an optimization (we convert provided ASTs back into
    * string).
    *
-   * The second object will be the `contextValue` object available in resolvers.
+   * `request.contextValue` will be the `contextValue` object available in resolvers.
    */
-  // TODO(AS4): document this
   public async executeOperation(
     this: ApolloServer<BaseContext>,
-    request: Omit<GraphQLRequest, 'query'> & {
-      query?: string | DocumentNode;
-    },
+    { request }: Omit<ExecuteOperation<TContext>, 'contextValue'>,
   ): Promise<GraphQLResponse>;
-  public async executeOperation(
-    request: Omit<GraphQLRequest, 'query'> & {
-      query?: string | DocumentNode;
-    },
-    contextValue: TContext,
-  ): Promise<GraphQLResponse>;
-
-  async executeOperation(
-    request: Omit<GraphQLRequest, 'query'> & {
-      query?: string | DocumentNode;
-    },
-    contextValue?: TContext,
-  ): Promise<GraphQLResponse> {
+  public async executeOperation({
+    request,
+    contextValue,
+  }: WithRequired<
+    ExecuteOperation<TContext>,
+    'contextValue'
+  >): Promise<GraphQLResponse>;
+
+  async executeOperation({
+    request,
```

Comment:
I sorta liked the mandatory operation object being an unnamed parameter, but maybe that's silly.
I also don't know if I love calling it request, which doesn't pair well with the name `executeOperation`, even though it is of request type... honestly I'd call this `executeParsedRequest` or something but prefer keeping the name matching the AS3 name.

In any case, there are docs showing executeOperation that need to be updated.

Reply:
We can deprecate to rename and keep as an alias for the life of AS4?

Reply:
We chatted on Slack and have decided to leave the method name as `executeOperation` but to move the operation itself back to being a first unnamed argument before the options object.